KubeArmor/core/dockerHandler.go

```diff
@@ -136,6 +147,59 @@ func (dh *DockerHandler) GetContainerInfo(containerID string) (tp.Container, err
 
 	// == //
 
+	if cfg.GlobalCfg.StateAgent && !cfg.GlobalCfg.K8sEnv {
+		container.ContainerImage = inspect.Config.Image //+ kl.GetSHA256ofImage(inspect.Image)
+
+		container.NodeName = cfg.GlobalCfg.Host
+
+		labels := []string{}
+		for k, v := range inspect.Config.Labels {
+			labels = append(labels, k+"="+v)
+		}
+
+		if _, ok := containerLabels["kubearmor.io/container.name"]; !ok {
+			labels = append(labels, "kubearmor.io/container.name="+container.ContainerName)
+		}
+
+		container.Labels = strings.Join(labels, ",")
+
+		var podIP string
+		if inspect.HostConfig.NetworkMode.IsNone() || inspect.HostConfig.NetworkMode.IsContainer() {
+			podIP = ""
+		} else if inspect.HostConfig.NetworkMode.IsHost() {
+			podIP = dh.NodeIP
+		} else if inspect.HostConfig.NetworkMode.IsDefault() {
+			podIP = inspect.NetworkSettings.Networks["bridge"].IPAddress
+		} else {
+			networkName := inspect.HostConfig.NetworkMode.NetworkName()
+			podIP = inspect.NetworkSettings.Networks[networkName].IPAddress
+		}
```

Comment:
suggestion: switch case would have been more readable

Reply:
@Ankurk99 That might not be possible as these functions return bools.